nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/action/ActionToFlowActionConverter.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.action;
+
+import org.apache.nifi.action.component.details.ComponentDetails;
+import org.apache.nifi.action.component.details.ExtensionDetails;
+import org.apache.nifi.action.component.details.RemoteProcessGroupDetails;
+import org.apache.nifi.action.details.ActionDetails;
+import org.apache.nifi.action.details.ConfigureDetails;
+import org.apache.nifi.action.details.ConnectDetails;
+import org.apache.nifi.action.details.MoveDetails;
+import org.apache.nifi.action.details.PurgeDetails;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Converts an {@link Action} to a {@link FlowAction} using {@link FlowActionAttribute} attributes.
+ */
+public class ActionToFlowActionConverter implements ActionConverter {
+
+    @Override
+    public FlowAction convert(Action action) {
+        Map<String, String> attributes = new HashMap<>();
+        attributes.putAll(actionAttributes(action));
+        attributes.putAll(actionDetailsAttributes(action.getActionDetails()));
+        attributes.putAll(componentDetailsProperties(action.getComponentDetails()));
```

Comment:
The individual methods create new Maps, but it would be more efficient to pass around the initial `attributes` Map to each method.